R/utils.R

```diff
@@ -365,3 +365,173 @@ filter_df_from_server_data <- function(
     return(data.frame())
   }
 }
+
+
+get_os <- function(){
+  sysinf <- Sys.info()
+  if (!is.null(sysinf)){
+    os <- sysinf['sysname']
+    if (os == 'Darwin')
+      os <- "macos"
+  } else { ## mystery machine
+    os <- .Platform$OS.type
+    if (grepl("^darwin", R.version$os))
+      os <- "macos"
+    if (grepl("linux-gnu", R.version$os))
+      os <- "linux"
+  }
+  tolower(os)
+}
+
+
+set_env_path <- function(os) {
+  exec_path <- fs::path("DYNAMO-HIA", switch(os,
+                                             windows = "DYNAMO-HIA.exe",
+                                             macos = fs::path("MacOS", "DYNAMO-HIA"),
+                                             linux = fs::path("bin", "DYNAMO-HIA"))
+  )
+
+  Sys.setenv(DYNAMO_HIA_PATH = exec_path)
+}
+
+
+#' Download and extract GitHub release asset
+#'
+#' @param repo_url Character string of the repository URL.
+#' Default is `"https://github.com/Dynamo-HIA/dynamo-hia"`.
+#' @param release_tag Character string of the release tag. Default is `"v3.0.0-beta.1"`.
+#' @param os Optional character string of the operating system. Default is `NULL` which tries
+#' to detect the OS automatically.
+#' @param dest_dir Character string of the destination directory for extraction
+#' (default: `tempdir()`).
+#' @param make_executable Logical indicating whether to set execute permissions. Default is `TRUE`.
+#' @param github_pat Optional GitHub Access Token. Default is `NULL` which will try to use the
+#' token from the environment variable `GITHUB_PAT`.
+#'
+#' @return An invisible `fs::path` object pointing to the extracted contents.
+#'
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' # With default arguments
+#' download_github_release()
+#'
+#' # From specific repository and release tag
+#' download_github_release(
+#'   repo_url = "https://github.com/Dynamo-HIA/dynamo-hia",
+#'   release_tag = "v3.0.0-beta.1"
+#' )
+#' }
+#'
+download_github_release <- function(repo_url = "https://github.com/Dynamo-HIA/dynamo-hia",
+                                    release_tag = "v3.0.0-beta.1",

```

Comment:
If possible, I would try to avoid hard-coding the release name here. depending on how frequently we expect new releases of HIA, we could (i) mention this explicitly somewhere in the developer documentation and/or (ii) store the variable explicitly somewhere in a "globals" file (as an env variable), or alternatively (iii) automatically get the name of the latest release from github?

Reply:
Hmm, if we require an env variable, this makes it more complicated for the user (especially nontechnical ones). The problem with using the latest release is that the function with the default values is not reproducible anymore, i.e., it might give different results at different times after a new release was made. I will mention it in the developer documentation.

Reply:
ok!